osu.Game/OsuGame.cs

```diff
@@ -655,15 +655,23 @@ public void PresentScore(IScoreInfo score, ScorePresentType presentType = ScoreP
                 Ruleset.Value = databasedScore.ScoreInfo.Ruleset;
                 Beatmap.Value = BeatmapManager.GetWorkingBeatmap(databasedBeatmap);
 
-                switch (presentType)
+                if (screen is SongSelect songSelect)
+                    songSelect.FinaliseSelection(databasedBeatmap, databasedScore.ScoreInfo.Ruleset, customStartAction: push);
+                else
+                    push();
```

Comment:
This thing is making it *very* painful for me to accept this fix. I understand the rationale for it, as well as the intention to eliminate this class of bugs entirely by using `FinaliseSelection()` everywhere, but this sort of thing spreading out to `OsuGame`, combined with the added complexity and another footgun to avoid in `SongSelect` (which already feels impossible to meaningfully touch, given how convoluted the decoupled bindables are and the carousel is) makes it much more difficult to swallow in my view.

Like at this point I would probably accept something much more direct over this, like something along the lines of:

```diff
diff --git a/osu.Game/Screens/Select/SongSelect.cs b/osu.Game/Screens/Select/SongSelect.cs
index c5e914b461..dcdab964eb 100644
--- a/osu.Game/Screens/Select/SongSelect.cs
+++ b/osu.Game/Screens/Select/SongSelect.cs
@@ -383,6 +383,12 @@ protected override IReadOnlyDependencyContainer CreateChildDependencies(IReadOnl
 
         public void Edit(BeatmapInfo? beatmapInfo = null)
         {
+            if (!Carousel.BeatmapSetsLoaded)
+            {
+                Logger.Log($"{nameof(Edit)} aborted as carousel beatmaps are not yet loaded");
+                return;
+            }
+
             if (!AllowEditing)
                 throw new InvalidOperationException($"Attempted to edit when {nameof(AllowEditing)} is disabled");
 ```

Probably would like a few other opinions to sound off on this (cc @ppy/team-client)

Reply:
That's fine, and was the initial solution I had. But I'm not anywhere close to 100% confident that other issues will arise from not running the `FinaliseSelection` flow when pushing a child screen to song select.